.github/workflows/nightly-release.yaml

```diff
@@ -0,0 +1,814 @@
+            if [[ ! -f "$file" ]]; then
+              echo "❌ Required file missing: $file"
+              exit 1
+            fi
+            echo "✅ Found: $file"
+          done
+
+          # Validate YAML syntax
+          if command -v python3 >/dev/null 2>&1; then
+            for file in "${required_files[@]}"; do
+              if python3 -c "import yaml; yaml.safe_load(open('$file'))" 2>/dev/null; then
+                echo "✅ Valid YAML: $file"
+              else
+                echo "❌ Invalid YAML: $file"
+                exit 1
+              fi
+            done
+          fi
+
+          echo "✅ Configuration validation passed"
+
+  release:
+    name: Nightly Release (K8s ${{ needs.pre-checks.outputs.kubernetes_version || 'v1.31.0' }})
+    runs-on: ubuntu-latest
+    needs: pre-checks
+    if: needs.pre-checks.outputs.should_release == 'true'
+
+    permissions:
+      contents: read
+      packages: write
+      id-token: write
+
+    strategy:
+      fail-fast: true
+      matrix:
+        kubernetes_version: [v1.31.0]
+
+    env:
+      KUBERNETES_VERSION: ${{ matrix.kubernetes_version }}
+      VERSION_TAG: ${{ needs.pre-checks.outputs.version_tag }}
+      LATEST_SHA: ${{ needs.pre-checks.outputs.latest_sha }}
+      IS_FORK: ${{ needs.pre-checks.outputs.is_fork }}
+      KO_EXTRA_ARGS: ${{ needs.pre-checks.outputs.ko_extra_args }}
+
+    steps:
+      - name: Release Information
+        run: |
+          set -euo pipefail
+
+          echo "🚀 Starting Tekton Nightly Release"
+          echo "=================================="
+          echo "Repository: ${{ github.repository }}"
+          echo "Mode: ${{ env.IS_FORK == 'true' && 'Fork Testing' || 'Production' }}"
+          echo "Kubernetes: ${{ env.KUBERNETES_VERSION }}"
+          echo "Version: ${{ env.VERSION_TAG }}"
+          echo "SHA: ${{ env.LATEST_SHA }}"
+          echo "Dry Run: ${{ env.DRY_RUN }}"
+          echo "Timestamp: $(date -u)"
+          echo "=================================="
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Full history for accurate versioning
+
+      - name: Set up Kind cluster
+        uses: helm/kind-action@v1.8.0
+        with:
+          node_image: kindest/node:${{ env.KUBERNETES_VERSION }}
+          cluster_name: tekton-nightly
+
+      - name: Verify Kubernetes cluster
+        run: |
+          set -euo pipefail
+
+          echo "🐳 Verifying Kubernetes cluster..."
+          echo "Kubernetes version: ${{ env.KUBERNETES_VERSION }}"
+
+          # Display cluster info
+          kubectl cluster-info
+          kubectl get nodes -o wide
+          kubectl get pods --all-namespaces
+
+          # Wait for cluster to be fully ready
+          kubectl wait --for=condition=Ready nodes --all --timeout=300s
+
+          echo "✅ Kubernetes cluster is ready"
+
+      - name: Install Tekton Pipelines and verify installation
```

Comment:
Don't we have this somewhere in some scripts in the repository ? If yes, we may want to re-use and share that code.